### PR TITLE
Strict typescript

### PR DESCRIPTION
**Pull request by moodysalem**

### CODE REVIEW

File: src/state/application/hooks.ts [hunk 1]
```diff
@@ -4,17 +4,17 @@ import { addPopup, PopupContent, removePopup, toggleWalletModal } from './action
 import { useSelector, useDispatch } from 'react-redux'
 import { AppState } from '../index'
 
-export function useBlockNumber() {
+export function useBlockNumber(): number | undefined {
   const { chainId } = useWeb3React()
 
-  return useSelector((state: AppState) => state.application.blockNumber[chainId])
+  return useSelector((state: AppState) => state.application.blockNumber[chainId ?? -1])
```
Comment (NoahZinsmeister) [inline src/state/application/hooks.ts:10]: hm the other changes make sense, but i don't see why this one is necessary.

Comment (moodysalem) [reply src/state/application/hooks.ts:10]: chainId can be `undefined`, and undefined can't be used to index into the object with numeric keys

Comment (NoahZinsmeister) [reply src/state/application/hooks.ts:10]: yeah i see after reading more, just feels a little funky to have the `-1` in there

Comment (NoahZinsmeister) [reply src/state/application/hooks.ts:10]: anyway it's fine, LGTM

**Pull request merged**